### DISCUSSION TITLE
Scaphandre vs Shelly Plug S

### DISCUSSION THREAD

**Discussion by itwars** [hubblo-org/scaphandre#338] [General]
I just got my new Shelly-Plug-S to have power consumption on all equipments and when measuring using both Shelly and Scaphandre there's some differences on regular bases:

- Shelly plug: 2.42W (https://www.it-wars.com/shelly.jpg)
- Scaphandre: 3.02W (https://www.it-wars.com/scaphandre.jpg)

It's not a huge difference of consumption on my minipc, but considering Scaphandre get metrics from CPU/RAM, but not drive, network and other stuffs, normally it should tell me less than 2.24W? Am I rigth?

BR

**Comment by PierreRust** [comment]
Scaphandre should definitely report an lower consumption that the Shelly Plug , as the plug measurement includes the motherboard, power supply loss, disks, etc.  
In your example, the consumption reported by the plug looks very low, even a mini pc usually uses way more than that.  As a comparison, I'm using a low power fanless celeron-based system and I still have around 10w when measuring at the plug level. What mini pc are you using, and do you read the plug measurements ?

**Comment by itwars** [comment]
Hello Pierre,

It's this computer: https://www.amazon.fr/gp/product/B08117XQ5X

- Quad-Core Intel Celeron N3450 Processor 
- 4G DDR4
- eMMC 64GB 
- plus an additional 256Go SSD

I just add 2 previews of grafana on the initial post!

On it there's 5 running kvm instances:
```
List VMs on Node n°3
 Id   Name                State
-----------------------------------
 1    srv-drone-agent-0   running
 2    srv-gitea-0         running
 3    srv-drone-cicd-0    running
 4    srv-grafana-0       running
 5    srv-prometheus-0    running
```

**Comment by PierreRust** [comment]
Well, hard to say, that's a very low power system indeed, but even though 2,5W seems too low imho. 
Maybe the shelly plug are not precise enough when measuring such low wattage :shrug:

**Comment by itwars** [comment]
Perhaps! It's based on ESP8266 but I can't fine any mention of power consumption? Did someone made same measurement with other devices but Shelly?

**Comment by bpetit** [comment]
Hi, thanks for this notice !

I'm about to acquire a "smart" wattmeter to do some comparisons as well. In a R&D project we compared scaphandre measurements with an IPMI card measurement, which shows what's expected : scaphandre gives a lower power usage than tha IPMI (which seems normal as RAPL only accounts the power used by CPU and RAM, not fans, hard drives, and so on... yet). 

You case as it would be the opposite, scaphandre giving a higher power figure.

Once I have some gear I'll run some tests on lower power/smaller devices and keep this thread posted.

Have a nice day,

**Comment by bpetit** [comment]
Playing with shelly plug, regular and new rapl metrics these days, more to come.

This will be a whole topic in itself as there are many uncertainties to deal with.

Efforts will be combined with the [Energizta](https://github.com/Boavizta/Energizta/) project which intends to give more clarity on the gaps between RAPL metrics and actual host power usage (same for IPMI).

I move this as a discussion as this will lead to many interesting discussions for sure.

![Capture d’écran du 2023-08-11 04-15-58](https://github.com/hubblo-org/scaphandre/assets/906428/d472e619-73bf-4c85-b7b2-ffe99456d995)

In short, from what seems pretty sure now, the shelly plug s does consume a bit a power by himself, but nor more than 0.5W as a peak from what I've seen. I don't know if it's enough to explain the initial issue, also because as you said several components are not taken into account by RAPL PKG (a bit more components are accounted for with RAPL PSYS, but this a fresh new metric, currently in dev, so I need more time to document it and talk more about it).

![Capture d’écran du 2023-08-08 13-25-32](https://github.com/hubblo-org/scaphandre/assets/906428/84d8085a-f566-4e0d-9da8-09ec123b1557)
